### PR TITLE
[BlockSparseArrays] Simplifications of `blocks` for blocksparse `Adjoint` and `Transpose`

### PR DESCRIPTION
**Pull request by lkdvos**
This PR excises `SparseTransposeBlocks` and `SparseAdjointBlocks`, and replaces them with the equivalent `adjoint(blocks(a))` and `transpose(blocks(a))`.
This should not change any functionality, but reduces the complexity as the number of wrapper types decreases.

### CODE REVIEW

File: NDTensors/src/lib/SparseArrayInterface/src/abstractsparsearray/wrappedabstractsparsearray.jl [hunk 1]
```diff
@@ -7,3 +7,13 @@ const WrappedAbstractSparseArray{T,N,A} = WrappedArray{
...
   <:AbstractSparseArray{T,N},<:WrappedAbstractSparseArray{T,N}
 }
+
+function stored_indices(a::Adjoint)
+  return Iterators.map(I -> CartesianIndex(reverse(I.I)), stored_indices(parent(a)))
+end
+nstored(a::Adjoint) = nstored(parent(a))
+
+function stored_indices(a::Transpose)
+  return Iterators.map(I -> CartesianIndex(reverse(I.I)), stored_indices(parent(a)))
```
Comment (mtfishman) [inline NDTensors/src/lib/SparseArrayInterface/src/abstractsparsearray/wrappedabstractsparsearray.jl:17]:
```suggestion
  return Iterators.map(I -> CartesianIndex(reverse(Tuple(I))), stored_indices(parent(a)))
```

**Comment by lkdvos** [comment]
This should 🤞 now have all necessary imports and be based on the latest main

File: NDTensors/src/lib/BlockSparseArrays/test/test_basics.jl [hunk 2]
```diff
@@ -210,6 +204,15 @@ using .NDTensorsTestUtils: devices_list, is_supported_eltype
...
     end
+
+    # adjoint and transpose
+    a = dev(BlockSparseArray{elt}([1], [1, 1]))
+    @allowscalar a[1, 2] = 1
+    @test [@views(a[it]) for it in block_stored_indices(a)] isa Vector
+    ah = adjoint(a)
+    @test [@views(ah[it]) for it in block_stored_indices(ah)] isa Vector
+    at = transpose(a)
+    @test [@views(at[it]) for it in block_stored_indices(at)] isa Vector
```
Comment (mtfishman) [inline NDTensors/src/lib/BlockSparseArrays/test/test_basics.jl:215]: Could you make these tests a bit more comprehensive, i.e. maybe set some blocks and then check they are set correctly, check the types of the blocks, etc.?

**Comment by mtfishman** [comment]
This is great, thanks!

**Comment by mtfishman** [comment]
@lkdvos did this fix: https://github.com/ITensor/ITensors.jl/issues/1336#issuecomment-2215122254?

**Comment by lkdvos** [comment]
Yeah, I double checked and it doesn't. I'll have a look if I can find a solution

**Pull request merged**